pretenders/server/apps/preset.py

```diff
@@ -89,6 +91,14 @@ def add_preset(uid):
         raise HTTPResponse(("Preset has {0} times. Must be greater than "
                              "zero.".format(preset.times).encode()),
                            status=400)
+    try:
+        pretender = get_pretenders('http')[uid]
+        if datetime.datetime.now() - pretender.start > pretender.timeout:
```

Comment:
I'd prefer to extract this logic into a function e.g. `pretender.has_expired` in `PretenderModel`.